aiocache/base.py

```diff
@@ -498,12 +489,13 @@ async def close(self, *args, _conn=None, **kwargs):
     async def _close(self, *args, **kwargs):
         pass
 
-    def _build_key(self, key, namespace=None):
-        if namespace is not None:
-            return "{}{}".format(namespace, _ensure_key(key))
-        if self.namespace is not None:
-            return "{}{}".format(self.namespace, _ensure_key(key))
-        return key
+    def build_key(self, key, namespace=None):
+        key_name = key.value if isinstance(key, Enum) else key
+        ns = namespace if namespace is not None else self.namespace
+        return self._build_key(key_name, namespace=ns)
+
+    def _build_key_default(self, key, namespace=None):
+        return "{}{}".format(namespace or "", key)
```

Comment:
This difference is meaningful at the level above this point (`BaseCache.build_key()`) because `namespace=None` will use the cache's default, whereas `namespace=""` is an override of the cache's default to use no namespace prefix.

I think you're right that at this level (`BaseCache._build_key()`) there is _**probably**_ no difference between `None` and `""`.  However, that does take away that particular freedom from the end user who might care about that distinction in their custom `key_builder`.

My inclination is slightly in favor of keeping this as is, but it's only based on _possibility_ rather than a concrete use case.

Reply:
Yep, but I can't think of any real cases for it, and ensuring it is always `str` means that the key builders all become simpler code and less likely that a user writes incorrect code (or has to special case `None` even though they will never use it).

Reply:
That makes sense. It might be the tiniest smidge faster also.

Let's give it a try.